cmd/wallet-sdk-js/jsinterop/agent.go

```diff
@@ -0,0 +1,59 @@
+//go:build js && wasm
+
+/*
+Copyright Gen Digital Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package jsinterop implements interops between wallet sdk and js.
+package jsinterop
+
+import (
+	"fmt"
+	"syscall/js"
+
+	"github.com/hyperledger/aries-framework-go/component/kmscrypto/kms"
+	"github.com/hyperledger/aries-framework-go/component/storage/indexeddb"
+
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-js/walletsdk"
+)
+
+const (
+	dbNamespace = "wallet-sdk"
+)
+
+var agentInstance *walletsdk.Agent
+var agentMethodsRunner AsyncRunner
+
+func InitAgent(_ js.Value, args []js.Value) (any, error) {
+	didResolverURI, err := ensureString(getOptionalNamedArgument(args, "didResolverURI"))
+	if err != nil {
+		return nil, err
+	}
+
+	if agentInstance != nil {
+		return nil, fmt.Errorf("agent intance already initialized")
+	}
+
+	indexedDBKMSProvider, err := indexeddb.NewProvider(dbNamespace)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create IndexedDB provider: %w", err)
+	}
+
+	kmsStore, err := kms.NewAriesProviderWrapper(indexedDBKMSProvider)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Aries KMS store wrapper")
```

Comment:
Missing error wrap here

Reply:
Error wrapping is another big topic for JS interop. Currently, it is not clear what best way to do it. Let's create TODO or Issue for this part.

Reply:
OK, though I still recommend adding the wrapping here now for consistency with all the other error returns until we figure out the best way to do it. It's just this one where the error isn't wrapped.